server/src/main/java/com/github/animelist/animelist/controller/RatingSystemController.java

```diff
@@ -0,0 +1,59 @@
+package com.github.animelist.animelist.controller;
+import com.github.animelist.animelist.model.JwtUserDetails;
+import com.github.animelist.animelist.model.input.RatingSystemInput;
+import com.github.animelist.animelist.model.input.RatingSystemType;
+import com.github.animelist.animelist.model.ratingsystem.ContinuousRatingSystem;
+import com.github.animelist.animelist.model.ratingsystem.DiscreteRatingSystem;
+import com.github.animelist.animelist.model.ratingsystem.RatingSystem;
+import com.github.animelist.animelist.service.RatingSystemService;
+import org.bson.types.ObjectId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.graphql.data.method.annotation.Argument;
+import org.springframework.graphql.data.method.annotation.MutationMapping;
+import org.springframework.graphql.data.method.annotation.QueryMapping;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.stereotype.Controller;
+
+import static com.github.animelist.animelist.util.AuthUtil.getUserDetails;
+
+@Controller
+public class RatingSystemController {
+    private final RatingSystemService ratingSystemService;
+
+    @Autowired
+    public RatingSystemController(
+            final RatingSystemService ratingSystemService) {
+        this.ratingSystemService = ratingSystemService;
+    }
+    @QueryMapping
+    @PreAuthorize("isAuthenticated()")
+    public RatingSystem getRatingSystem(@Argument("ratingSystemID") final String ratingSystemID){
+        return ratingSystemService.getRatingSystem(ratingSystemID).orElseThrow();
+    }
+
+    @MutationMapping
+    @PreAuthorize("isAuthenticated()")
+    public RatingSystem createRatingSystem(@Argument("input") final RatingSystemInput input){
+        JwtUserDetails userDetails = getUserDetails();
+
+        final RatingSystem ratingSystem;
+        if (input.ratingSystemType() == RatingSystemType.DISCRETE) {
```

Comment:
style change can be implemented using the switch pattern matching in newer versions of java
```
return switch(input.ratingSystemType()) {
    case DISCRETE -> DiscreteRatingSystem.builder()...
    case CONTINUOUS -> ContinuousRatingSystem.builder()...
}
```